tests/integration/test_application_metadata_view.py

```diff
@@ -0,0 +1,227 @@
+from uuid import UUID
+
+import pytest
+from katka import models
+
+
+@pytest.mark.django_db
+class TestApplicationMetadataList:
+    def test_unauthenticated(self, client, application, metadata):
+        url = f'/applications/{application.public_identifier}/metadata/'
+        response = client.get(url)
+        assert response.status_code == 200
+        parsed = response.json()
+        assert len(parsed) == 0
+
+    def test_authenticated(self, client, logged_in_user, application, metadata):
+        response = client.get(f'/applications/{application.public_identifier}/metadata/')
+        assert response.status_code == 200
+        parsed = response.json()
+        assert len(parsed) == 1
+        metadata = parsed[0]
+        assert metadata['key'] == 'ci'
+        assert metadata['value'] == 'the-team'
+        assert UUID(metadata['application']) == application.public_identifier
+
+
+@pytest.mark.django_db
+class TestApplicationMetadataGet:
+    def test_unauthenticated(self, client, application, metadata):
+        url = f'/applications/{application.public_identifier}/metadata/{metadata.key}'
+        response = client.get(url)
+        assert response.status_code == 404
+
+    def test_authenticated(self, client, logged_in_user, application, metadata):
+        url = f'/applications/{application.public_identifier}/metadata/{metadata.key}/'
+        response = client.get(url)
+        assert response.status_code == 200
+        metadata = response.json()
+        assert metadata['key'] == 'ci'
+        assert metadata['value'] == 'the-team'
+        assert UUID(metadata['application']) == application.public_identifier
+
+    def test_non_existent_application(self, client, logged_in_user, application, metadata):
+        url = f'/applications/00000000-0000-0000-0000-000000000000/metadata/{metadata.key}/'
+        response = client.get(url)
+        assert response.status_code == 404
+
+    def test_non_existent_key(self, client, logged_in_user, application, metadata):
+        url = f'/applications/{application.public_identifier}/metadata/wrong_key/'
+        response = client.get(url)
+        assert response.status_code == 404
+
+    def test_not_my_metadata(self, client, logged_in_user, not_my_application, not_my_metadata):
+        url = f'/applications/{not_my_application.public_identifier}/metadata/{not_my_metadata.key}/'
+        response = client.get(url)
+        assert response.status_code == 404
```

Comment:
I guess this happens everywhere so far, but should this be a 404? and not a 403?

Reply:
it's a 404 because you don't know that a given data from someone else exists. If you actually return 403 you will acknowledge that a wrong data exists but it's not yours. So, this can give a lot of insights for malicious people